chat-plugins/roomsettings.js

```diff
@@ -59,7 +59,10 @@ class RoomSettings {
 		return modchatOutput.join(' ');
 	}
 	modjoin() {
-		if (!this.user.can('makeroom') && !this.room.isPersonal) return this.button(this.room.modjoin ? this.room.modjoin : 'off', true);
+		if (!this.user.can('makeroom') && !this.room.isPersonal ||
+			!this.user.can('editroom', null, this.room)) {
```

Comment:
Anyway, >80col lines should be the exception, not the rule. I try to fix them whenever I see them, but not _too_ hard since I generally consider functionality more important.